go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-operator
 
-go 1.22.0
+go 1.23
+
+toolchain go1.23.2
```

Comment:
FYI, as per the Go support policy, we can't set 1.23 as the minimum until 1.24 is released.